core/primitives/src/types.rs

```diff
@@ -146,6 +146,11 @@ pub type MaybeBlockId = Option<BlockId>;
 
 #[derive(Default, BorshSerialize, BorshDeserialize, Clone, Debug, Eq, PartialEq)]
 pub struct ValidatorStats {
-    pub blocks_produced: BlockIndex,
-    pub blocks_expected: BlockIndex,
+    pub produced: BlockIndex,
+    pub expected: BlockIndex,
+}
+
+pub struct BlockChunkValidatorStats {
+    pub block_stats: ValidatorStats,
+    pub chunk_stats: ValidatorStats,
```

Comment:
This structure does not make any sense to me even after I read the code where it is used. I don't understand why block and chunk stats are on the same level. When those can be different and why?

Reply:
This is stats how many blocks this validator produced and how many chunks this validator produced.
Why would it be on different level? They are always different because validators are expected to produce epoch_length/num_validators and epoch_length/(num_validators/num_shards) of chunks.